pytorch_lightning/utilities/device_dtype_mixin.py

```diff
@@ -132,7 +132,8 @@ def cuda(self, device: Optional[int] = None) -> Module:
         Returns:
             Module: self
         """
-        self.__update_properties(device=torch.device('cuda', index=device))
+        property_device = device if isinstance(device, torch.device) else torch.device('cuda', index=device)
+        self.__update_properties(device=property_device)
```

Comment:
Let's make a test for this in particular, DP was passing in torch devices which using the standard `cuda` function work fine, but for our device mixins, they were unsupported which raised an error once we are not moving devices correctly.

Reply:
We have tests/utilities/test_dtype_device_mixin.py, we can add it there.